packages/near-membrane-base/src/red.ts

```diff
@@ -183,11 +178,16 @@ export const serializedRedEnvSourceText = (function redEnvFactory(blueEnv: Membr
     }
 
     function getDistortedValue(target: RedProxyTarget): RedProxyTarget {
-        if (!WeakMapHas(distortionMap, target)) {
-            return target;
+        let distortedTarget: RedProxyTarget = target;
+        try {
+            distortedTarget = distortionCallback(target);
+        } finally {
+            // if a distortion entry is found, it must be a valid proxy target
+            if (distortedTarget !== target && typeof distortedTarget !== typeof target) {
```

Comment:
is `typeof distortedTarget !== typeof target` to allow `NaN`?

Reply:
well, now that you mentioned, this is wrong because it doesn't cover the case when the callback throws... 

but in general, thsi is the same check we have in the old code during the ingestion of the distortion map, checking that a distortion for a function correspond to a function, etc.